### PR TITLE
unity: add handling fails without message in harness

### PR DESCRIPTION
**Pull request by adamdebek**
JIRA: CI-464

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: trunner/harness/unity.py [hunk 1]
```diff
@@ -27,14 +27,15 @@ def unity_harness(dut: Dut, ctx: TestContext, result: TestResult) -> Optional[Te
...
-                parsed["msg"] = last_assertion["msg"]
+                msg = last_assertion["msg"]
                 last_assertion = {}
 
             status = Status.from_str(parsed["status"])
             subname = f"{parsed['group']}.{parsed['name']}"
             if "path" in parsed and "line" in parsed:
-                parsed["msg"] = f"[{parsed['path']}:{parsed['line']}] " + parsed["msg"]
+                parsed["msg"] = f"[{parsed['path']}:{parsed['line']}] " + msg
             result.add_subresult(subname, status, parsed.get("msg", ""))
```
Comment (mateusz-bloch) [inline trunner/harness/unity.py:39]: don't need to use `.get` when we have declared msg earlier as empty, also can put everything into a formatted string
```suggestion
                msg = f"[{parsed['path']}:{parsed['line']}] {msg}"
            result.add_subresult(subname, status, msg)
```

File: trunner/harness/unity.py [hunk 2]
```diff
@@ -27,14 +27,15 @@ def unity_harness(dut: Dut, ctx: TestContext, result: TestResult) -> Optional[Te
...
             if last_assertion:
-                parsed["msg"] = last_assertion["msg"]
+                msg = last_assertion["msg"]
                 last_assertion = {}
 
             status = Status.from_str(parsed["status"])
             subname = f"{parsed['group']}.{parsed['name']}"
             if "path" in parsed and "line" in parsed:
-                parsed["msg"] = f"[{parsed['path']}:{parsed['line']}] " + parsed["msg"]
+                parsed["msg"] = f"[{parsed['path']}:{parsed['line']}] " + msg
```
Comment (nalajcie) [inline trunner/harness/unity.py:38]: - this means that test can fail without any assert? 
 - please note that you're changing the semantics (the msg would be added to subresult only if `path` and `line` are present - not sure if relevant. Maybe just make a minor change (without the changes above):
```suggestion
                parsed["msg"] = f"[{parsed['path']}:{parsed['line']}] " + parsed.get("msg", "")
```

Comment (mateusz-bloch) [reply trunner/harness/unity.py:38]: Hmm, from what I understand now, the issue is that the test might fail with an assert, but the regex won't catch it due to the lack of message on the end of assert. In that case, I would adjust the regex, that this `: (?P<msg>.*?)` can be optional

Comment (adamdebek) [reply trunner/harness/unity.py]: Test can fail without any message, for example using `TEST_FAIL()` and in this situation there was an error:
image

Comment (nalajcie) [reply trunner/harness/unity.py]: I'm not sure writing tests which can fail without an explicit reason should be encouraged.

Still, adding `parsed.get("msg", "")` probably saves the day?

Comment (adamdebek) [reply trunner/harness/unity.py]: Yea, I agree, but I think such a case should be taken into account and handled. I used your suggestion and also adjusted assert regex.

**Review by damianloew — APPROVED** [review]

**Pull request merged**